scripts/Diffraction/isis_powder/pearl_routines/fmode_output.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import (absolute_import, division, print_function)
+
+import mantid.simpleapi as mantid
+
+import isis_powder.routines.common as common
+
+# This file generates the various outputs for the PEARL instruments and saves them to their respective files
+
+
+def generate_and_save_focus_output(instrument, processed_spectra, run_details, perform_attenuation, focus_mode=None):
+    output_file_paths = instrument._generate_out_file_paths(run_details=run_details)
+    cycle_information = instrument._get_label_information(run_details.run_number)
+    unused, save_range = instrument._get_instrument_alg_save_ranges(cycle_information["instrument_version"])
+
+    if focus_mode == "all":
+        processed_nexus_files = _focus_mode_all(output_file_paths, processed_spectra)
+
+    elif focus_mode == "groups":
+        processed_nexus_files = _focus_mode_groups(cycle_information, output_file_paths, save_range,
+                                                   processed_spectra)
+
+    elif focus_mode == "trans":
+
+        processed_nexus_files = _focus_mode_trans(output_file_paths, perform_attenuation, instrument, processed_spectra)
+
+    elif focus_mode == "mods":
+
+        processed_nexus_files = _focus_mode_mods(output_file_paths, processed_spectra)
+
+    else:
+        raise ValueError("Focus mode unknown")
+
+    return processed_nexus_files
+
+
+def _focus_mode_mods(output_file_paths, calibrated_spectra):
+    index = 1
+    append = False
+    output_list = []
+    for ws in calibrated_spectra:
+
+        mantid.SaveGSS(InputWorkspace=ws, Filename=output_file_paths["gss_filename"], Append=append, Bank=index)
+        output_name = output_file_paths["output_name"] + "_mod" + str(index)
+        dspacing_ws = mantid.ConvertUnits(InputWorkspace=ws, OutputWorkspace=output_name, Target="dSpacing")
+        output_list.append(dspacing_ws)
+        mantid.SaveNexus(Filename=output_file_paths["nxs_filename"], InputWorkspace=dspacing_ws, Append=append)
+
+        append = True
+        index += 1
+    return output_list
+
+
+def _focus_mode_trans(output_file_paths, atten, instrument, calibrated_spectra):
+    summed_ws = mantid.CloneWorkspace(InputWorkspace=calibrated_spectra[0])
+    for i in range(1, 9):  # Add workspaces 2-9 to workspace 1
+        summed_ws = mantid.Plus(LHSWorkspace=summed_ws, RHSWorkspace=calibrated_spectra[i])
```

Comment:
Good point I didn't realise there was an algorithm. Added as part of the refactoring work [here](https://github.com/mantidproject/mantid/issues/18108)